src/main/ruby/truffleruby/core/io.rb

```diff
@@ -857,6 +857,7 @@ def initialize(fd, mode = nil, **options)
 
     @autoclose = autoclose_tmp
     @pipe = false
+    @path = path unless @path
```

Comment:
This feels hacky, there should be no ` unless @path` here.
You should pass the path there to as a keyword argument to `super`.